utilities/write_batch_with_index/write_batch_with_index_internal.h

```diff
@@ -388,12 +391,45 @@ class WriteBatchWithIndexInternal {
   static Status MergeKeyWithNoBaseValue(ColumnFamilyHandle* column_family,
                                         const Slice& key,
                                         const MergeContext& context,
-                                        std::string* result);
+                                        std::string* result,
+                                        ValueType* result_type);
 
   static Status MergeKeyWithPlainBaseValue(ColumnFamilyHandle* column_family,
                                            const Slice& key, const Slice& value,
                                            const MergeContext& context,
-                                           std::string* result);
+                                           std::string* result,
+                                           ValueType* result_type);
+
+  static Status MergeKeyWithWideColumnBaseValue(
+      ColumnFamilyHandle* column_family, const Slice& key, const Slice& value,
+      const MergeContext& context, std::string* result, ValueType* result_type);
+
+  static Status MergeKeyWithWideColumnBaseValue(
+      ColumnFamilyHandle* column_family, const Slice& key,
+      const WideColumns& columns, const MergeContext& context,
+      std::string* result, ValueType* result_type);
+
+  static Status MergeKeyWithNoBaseValue(ColumnFamilyHandle* column_family,
```

Comment:
nit: group them by names? `MergeKeyWithNoBaseValue`, `MergeKeyWithPlainBaseValue`, and `MergeKeyWithWideColumnBaseValue`

Reply:
Sure, that ordering works too :)

Reply:
I ended up templatizing this part; that eliminated a whole bunch of duplicated code.

Reply:
Even better. Thank you!